framework/source/class/qx/core/Object.js

```diff
@@ -230,15 +230,34 @@ qx.Class.define("qx.core.Object",
     /**
      * Load user defined data from the object
      *
-     * @param key {String} the key
-     * @return {Object} the user data
+     * @param key {String?} the key
+     *   If key is not a string, pass the user the entire map
```

Comment:
why check for it being a string, so if one enters a number the full data is returned .. that seems odd ... shouldn't it be that only if 'null' is entered that the full data is returned.

Reply:
Even though the docstring specifies that the parameter is a string? Changes made.

Reply:
then you would have to error out if someone entered something other than a string ... I think with this change we are good.